packages/autocomplete-plugin-tags/src/createTagsPlugin.tsx

```diff
@@ -0,0 +1,173 @@
+/** @jsx createElement */
+import { BaseItem, PluginSubscribeParams } from '@algolia/autocomplete-core';
+import {
+  AutocompletePlugin,
+  AutocompleteSource,
+  AutocompleteState,
+} from '@algolia/autocomplete-js';
+
+import { createTags, OnTagsChangeParams } from './createTags';
+import type { BaseTag, Tag } from './types';
+
+type OnChangeParams<TTag> = PluginSubscribeParams<any> &
+  OnTagsChangeParams<TTag>;
+
+type TagsPluginData<TTag> = {
+  /**
+   * Returns the current list of tags.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-tags
+   */
+  readonly tags: Array<Tag<TTag>>;
+  /**
+   * Adds tags to the list.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-addtags
+   */
+  addTags: (tags: Array<BaseTag<TTag>>) => void;
+  /**
+   * Sets the list of tags.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-settags
+   */
+  setTags: (tags: Array<BaseTag<TTag>>) => void;
+};
+
+export type CreateTagsPluginParams<
+  TTag extends Record<string, any>,
+  TItem extends BaseItem
+> = {
+  /**
+   * A set of initial tags to pass to the plugin.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-initialtags
+   */
+  initialTags?: Array<BaseTag<TTag>>;
+  /**
+   * A function to specify what sources the plugin should subscribe to. The plugin adds a tag when selecting an item from these sources.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-gettagssubscribers
+   */
+  getTagsSubscribers?(): Array<{
+    sourceId: string;
+    getTag(params: { item: TItem }): BaseTag<TTag>;
```

Comment:
Would it make sense to also pass the `source` and the `state` here?

Reply:
What use cases do you have in mind?

Reply:
When using other plugins, I sometimes had to trigger different behaviors depending on the `sourceId` or on the current state (`state.context` for instance). Wondering if it would also be useful here. I haven't used the Tags API yet so I don't know yet.

Reply:
The purpose of this function is to map an item into a tag, there shouldn't be complicated logic. I think unless we face a scenario where `source` and `state` are needed, we should wait. This will be easy enough to add later on.